### ISSUE TITLE
unexpected keyword argument when adding schema to API Gateway model

### ISSUE THREAD

**Issue by JamesBelchamber** [aws/aws-cdk#5272]
When trying to add a Model to an API Gateway instance I'm getting an error. Providing an empty schema (`{}`) works as expected.

### Reproduction Steps

This is being added to a `LambdaRestApi`:

```
apigw.add_model(
    'empty_get_body',
    content_type="application/json",
    model_name="EmptyGetBody",
    schema={
        "$schema": "http://json-schema.org/draft-04/schema#",
        "title" : "Empty Schema",
        "type" : "object"
    }
)
```

### Error Log

Running `cdk synth`:

```
Traceback (most recent call last):
  File "app.py", line 47, in <module>
    env={'region': environ['AWS_DEFAULT_REGION'], 'account': environ['AWS_ACCOUNT_ID']})
  File "/Users/james.belchamber/Library/Python/3.7/lib/python/site-packages/jsii/_runtime.py", line 66, in __call__
    inst = super().__call__(*args, **kwargs)
  File "<<module-path-redacted>>", line 130, in __init__
    "type" : "object"
  File "/Users/james.belchamber/Library/Python/3.7/lib/python/site-packages/aws_cdk/aws_apigateway/__init__.py", line 13433, in add_model
    props = ModelOptions(schema=schema, content_type=content_type, description=description, model_name=model_name)
  File "/Users/james.belchamber/Library/Python/3.7/lib/python/site-packages/aws_cdk/aws_apigateway/__init__.py", line 12233, in __init__
    if isinstance(schema, dict): schema = JsonSchema(**schema)
TypeError: __init__() got an unexpected keyword argument '$schema'
```

### Environment

  - **CLI Version      :** 1.18.0 (build bc924bc)
  - **Framework Version:** 1.18.0
  - **OS               :** macOS Catalina (10.15.1)
  - **Language         :** Python (3.7.5)

This is :bug: Bug Report

**Comment by RomainMuller** [comment]
You need to remove the `$` in front of `$schema`. The accepted type for the `schema` property is [`JsonSchema`](https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.aws_apigateway/JsonSchema.html). You should also use one of the JSON schema version constants available on [`JsonSchemaVersion`](https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.aws_apigateway/JsonSchemaVersion.html#aws_cdk.aws_apigateway.JsonSchemaVersion).

**Comment by JamesBelchamber** [comment]
The documentation is _very_ hard to follow on this! I landed up following the notes in the code, but thinking about it those notes might be for the JS version of the codebase. I'll test this now, thanks :)
